cpp/src/io/parquet/stats_filter_helpers.cpp

```diff
@@ -117,12 +126,22 @@ std::reference_wrapper<ast::expression const> stats_expression_converter::visit(
   auto const op       = expr.get_operator();
 
   if (auto* v = dynamic_cast<ast::column_reference const*>(&operands[0].get())) {
-    // First operand should be column reference, second should be literal.
-    CUDF_EXPECTS(cudf::ast::detail::ast_operator_arity(op) == 2,
-                 "Only binary operations are supported on column reference");
-    CUDF_EXPECTS(dynamic_cast<ast::literal const*>(&operands[1].get()) != nullptr,
+    // First operand should be column reference, second (if binary operation) should be literal.
+    CUDF_EXPECTS(cudf::ast::detail::ast_operator_arity(op) == 1 or
+                   cudf::ast::detail::ast_operator_arity(op) == 2,
+                 "Only unary and binary operations are supported on column reference");
+    CUDF_EXPECTS(cudf::ast::detail::ast_operator_arity(op) == 1 or
+                   dynamic_cast<ast::literal const*>(&operands[1].get()) != nullptr,
                  "Second operand of binary operation with column reference must be a literal");
     v->accept(*this);
+
+    // For unary operators, push and return the `_always_true` expression
+    if (cudf::ast::detail::ast_operator_arity(op) == 1) {
+      _stats_expr.push(ast::operation{ast_operator::IDENTITY, *_always_true});
+      // Propagate the `_always_true` as expression to its unary operator parent
+      return *_always_true;
```

Comment:
@davidwendt Added a stream test that calls this code path. Interestingly, the test doesn't fail even when `_always_true` is constructed without a `stream`, not fully sure why.

I can't unfortunately add a small enough to test the identical code sections in `dictionary_page_filter.cu` and `bloom_filter_reader.cu`